### ISSUE TITLE
Non-existent template file gives 404 error, should be 500

### ISSUE THREAD

**Issue by DrHyde** [mojolicious/mojo#1728]
* Mojolicious version: 8.73
* Perl version: 5.30.1
* Operating system: Mac OS Catalina

### Steps to reproduce the behavior

Run this and then visit http://127.0.0.1:3000/

```perl
use Mojolicious::Lite -signatures;

get '/' => sub ($c) {
  print STDERR "Look ma, this route exists!\n";
  $c->render(
      template  => 'no/such/file',
      handler   => 'ep',
      format    => 'html',
      some_data => 'i like bubbles'
  );
};

app->start;
```

### Expected behavior

A 500 error

### Actual behavior

A 404 error

404 means that the _requested_ resource, ie `/`, could not be found, but it clearly *is* found, as the `print` to `STDERR` happens. A [500 error](https://tools.ietf.org/html/rfc7231#section-6.6.1) is more appropriate as it signals that the route was found but is broken.

**Comment by jberger** [comment]
This behavior was fixed in 9.0. Please upgrade, test, and reopen if it still doesn't meet your expectations at that point.

**Comment by DrHyde** [comment]
Ahh, thanks. Confirmed fixed in 9.01.
